tests/models/test_hparams.py

```diff
@@ -29,20 +30,44 @@
 from tests.base import EvalModelTemplate, TrialMNIST, BoringModel
 
 
-class SaveHparamsModel(EvalModelTemplate):
+class SaveHparamsModel(BoringModel):
     """ Tests that a model can take an object """
     def __init__(self, hparams):
         super().__init__()
         self.save_hyperparameters(hparams)
 
 
-class AssignHparamsModel(EvalModelTemplate):
+class AssignHparamsModel(BoringModel):
     """ Tests that a model can take an object with explicit setter """
     def __init__(self, hparams):
         super().__init__()
         self.hparams = hparams
 
 
+def decorate(func):
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        return func(*args, **kwargs)
+
+    return wrapper
+
+
+class SaveHparamsDecoratedModel(BoringModel):
+    """ Tests that a model can take an object """
+    @decorate
+    def __init__(self, hparams, *my_args, **my_kwargs):
+        super().__init__()
+        self.save_hyperparameters(hparams)
+
+
+class AssignHparamsDecoratedModel(BoringModel):
+    """ Tests that a model can take an object with explicit setter"""
+    @decorate
```

Comment:
Can you try with multiple decorator ? Would undecorated help there  ?

Reply:
https://github.com/PyTorchLightning/pytorch-lightning/blob/52a82394d531ad5f6f71327a8194632b06935e1a/tests/backends/launcher.py#L76

Reply:
added double decoration and works...
it may work but it does not simplify the getting keys words :]